src/main/java/org/example/homework4/LeapYear.java

```diff
@@ -0,0 +1,22 @@
+package org.example.homework4;
+
+public class LeapYear {
+    public static void main(String[] args) {
+        int yearValue = 2024;
+        System.out.println(yearValue + ": Is it a leap year?" + "\nAnswer: " + isLeapYear(yearValue));
+
+    }
+
+    public static boolean isLeapYear(int year) {
+        if (year < 1 || year > 9999) {
+            System.out.println("Year value is invalid");
+            return false;
+        } else if (year % 4 == 0 && year % 100 != 0) {
```

Comment:
You can combine (year % 4 == 0 && year % 100 != 0) and (year % 100 == 0 && year % 400 == 0) expressions into one condition.
In my opinion, there is no sense to split it. Because in both cases result is identical: return true. And because both of them are checking is it a leap year number